public/app/core/components/VersionHistory/VersionHistoryComparison.tsx

```diff
@@ -0,0 +1,103 @@
+import { identity } from 'lodash';
+import { useState } from 'react';
+
+import { dateTimeFormatTimeAgo } from '@grafana/data';
+import { Box, Button, Divider, EmptyState, Icon, Stack, Text } from '@grafana/ui';
+import { t, Trans } from 'app/core/internationalization';
+import { DiffGroup } from 'app/features/dashboard-scene/settings/version-history/DiffGroup';
+import { DiffViewer } from 'app/features/dashboard-scene/settings/version-history/DiffViewer';
+import { jsonDiff } from 'app/features/dashboard-scene/settings/version-history/utils';
+
+/** Meta information about a version of an entity */
+export interface RevisionModel {
+  version: number | string;
+  /** When was this version created? */
+  created: string;
+  /** Who created/edited this version? */
+  createdBy: string;
+  /** Optional message describing change encapsulated in this version */
+  message?: string;
+}
+
+type DiffArgument = Parameters<typeof jsonDiff>[0];
+
+type DiffViewProps<T extends DiffArgument> = {
+  /** Information to help summarise the change in the newer version */
+  newSummary: RevisionModel;
+  /** Information to help summarise the change in the older version */
+  oldSummary: RevisionModel;
+  /** The actual data model of the older version */
+  oldVersion: T;
+  /** The actual data model of the newer version */
+  newVersion: T;
+  /**
+   * Helper method to tweak the calculated diff for the human readable output.
+   *
+   * e.g. mapping machine IDs to translated names, removing fields that the user can't control anyway etc.
+   */
+  preprocessVersion?: (version: T) => DiffArgument;
+};
+
+const VersionChangeSummary = ({ info }: { info: RevisionModel }) => {
+  const { created, createdBy, version, message = '' } = info;
+  const ageString = dateTimeFormatTimeAgo(created);
+  return (
+    <Trans i18nKey="core.versionHistory.comparison.header.text">
+      Version {{ version }} updated by {{ createdBy }} ({{ ageString }}) {{ message }}
+    </Trans>
+  );
+};
+
+export const VersionHistoryComparison = <T extends DiffArgument>({
+  oldSummary,
+  newSummary,
+  oldVersion,
+  newVersion,
+  preprocessVersion = identity,
+}: DiffViewProps<T>) => {
+  const diff = jsonDiff(preprocessVersion(oldVersion), preprocessVersion(newVersion));
+  const noHumanReadableDiffs = Object.entries(diff).length === 0;
```

Comment:
Some properties always change between versions. We can process the diff and work out two comparisons - the human readable summary, and the raw JSON diff